### PR TITLE
Refactoring of the tutorial test cases, esp. in Java.

### PR DESCRIPTION
**Pull request by joeldudleyr3**

### CODE REVIEW

File: java-source/src/test/java/com/example/contract/IOUFlowTests.java [hunk 1]
```diff
@@ -41,44 +44,34 @@ public void tearDown() {
         net.stopNodes();
     }
 
-    @Test
-    public void flowRejectsInvalidIOUs() throws InterruptedException {
+    @Test(expected=ExecutionException.class)
```
Comment (shamsasari) [inline java-source/src/test/java/com/example/contract/IOUFlowTests.java]: Why was this changed? You're no longer checking for a specific exception

Comment (joeldudleyr3) [inline java-source/src/test/java/com/example/contract/IOUFlowTests.java]: Because the current structure is very confusing. Java has no "failsWith()" method, so you have to do a try/catch, catch the exception, then confirm its type. You fail the test by NOT throwing an exception, which is very confusing to people.

Comment (shamsasari) [inline java-source/src/test/java/com/example/contract/IOUFlowTests.java]: Fair enough I can see that. In that case we need to be doing assertions on exceptions properly - either use what's available in JUnit or use something like assertj.
